### PR TITLE
Fix github actions so they can be run locally

### PR DESCRIPTION
**Pull request by CyberGsus**
I tried running [`act`](https://github.com/nektos/act) to help with a PR and I encountered that the 
containers were not running correctly due to various errors. I've gone through each workflow and
made sure they ran. I also added a script to avoid repeating myself on the downloading and execution of
the appimage

- fix(CONTRIBUTING): Fixed a typo on CONTRIBUTING file
- fix(workflows/lint): Fixed `apt-get` hanging on confirm
- fix(workflows/ci): Updating APT repositories so it can find `fd-find`
- fix(workflows/ci/appimage-ubuntu): Tests for AppImage also run
- maint(workflows): Common script for setting up the appimage
- maint(workflows): Removed unused `fuse` dependency.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -46,12 +47,10 @@ jobs:
...
         run: |
-          sudo apt install fd-find
+          sudo apt update
+          sudo apt install fd-find --yes
           test -d build || {
-            mkdir -p build
-            wget https://github.com/neovim/neovim/releases/download/nightly/nvim.appimage
-            chmod +x nvim.appimage
-            mv nvim.appimage ./build/nvim
+            bash .github/workflows/scripts/setup-appimage.bash
```
Comment (Conni2461) [inline .github/workflows/ci.yml:53]: you already have `bash` in the shebang? why not make the script executable and do `./.github/workflows/scripts/setup-appimage.bash`

i also prefer scripts to end on `.sh`

File: .github/workflows/scripts/setup-appimage.bash [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+chmod +x nvim.appimage
+mv nvim.appimage ./build
+# AppImages require FUSE to run: https://docs.appimage.org/user-guide/troubleshooting/fuse.html#setting-up-fuse-on-ubuntu-debian-and-their-derivatives            # NOTE: afaik contaniers can't modprobe
+# and afaik docker containers can't run kernel mdoules.
+# XXX: Should we create a script somewhere... instead of this hacky workaround??
+echo "
+#!/bin/bash
+./build/nvim.appimage --appimage-extract-and-run \"\$@\"
+" > ./build/nvim
+chmod +x ./build/nvim
```
Comment (Conni2461) [inline .github/workflows/scripts/setup-appimage.bash:15]: Why do we need this part, i dont really understand it.

The current code already works perfectly fine, in github actions?!

I don't know act or whats its purpose so

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -58,7 +58,7 @@ The other option would be setting up https://github.com/tjdevries/tree-sitter-lu
 - Install Treesitter, either with package manager or with github release
 - Install plugin as usual
 - cd to plugin
-- `mkdir -p build parser` sadly does doesn't exist
+- `mkdir -p build parser` sadly those don't exist
```
Comment (Conni2461) [inline CONTRIBUTING.md:61]: Shouldn't be here, can you rebase? The first commit should not be part of this PR

File: .github/workflows/ci.yml (hunk 1 above)
Comment (CyberGsus) [reply .github/workflows/ci.yml:53]: Sure! I'll do this when I get home

File: .github/workflows/scripts/setup-appimage.bash (hunk 2 above)
Comment (CyberGsus) [reply .github/workflows/scripts/setup-appimage.bash:15]: Running the appimage directly requires FUSE, which needs to install a kernel module and can't be run on containers. The --appimage-extract-and-run flag enables executing the appimage without FUSE by extracting and instantly running. I had to male a wrapper script so that `make test` starts the appimage correctly

Comment (CyberGsus) [reply .github/workflows/scripts/setup-appimage.bash:15]: There are pull requests that have failed actions and logs not available due to a long time having passed. Unless the contributor has write access or atleast access to running the actions they cannot know the problem. Being able to run failed actions locally enables us to get progress done in pull requests that may have been abandoned when they were approved and the only thing that failed were lints

File: CONTRIBUTING.md (hunk 3 above)
Comment (CyberGsus) [reply CONTRIBUTING.md:61]: Sorry, I don't understand. Do you mean that this was already fixed? In that case, we can close this.

Comment (CyberGsus) [reply CONTRIBUTING.md:61]: Nvm, I understood. I thought I had submitted a separate PR. I'll revert these changes.

File: .github/workflows/scripts/setup-appimage.bash (hunk 2 above)
Comment (CyberGsus) [reply .github/workflows/scripts/setup-appimage.bash:15]: Act is this: https://github.com/nektos/act. It is a program that you install on your machine, reads the workflows directory and launches docker containers to execute the actions.

Comment (CyberGsus) [reply .github/workflows/scripts/setup-appimage.bash]: The pull request that had failed actions with expired logs is #663, I wanted to fix the lints but I couldn't see which lints didn't pass... I tried to run it locally, it hung in the yes/no prompt from apt, I fixed that, and I then took some care of the rest, know that they were up and so on.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -46,12 +47,11 @@ jobs:
...
-          sudo apt install fd-find
+          sudo apt update
+          sudo apt install fd-find --yes
           test -d build || {
-            mkdir -p build
-            wget https://github.com/neovim/neovim/releases/download/nightly/nvim.appimage
-            chmod +x nvim.appimage
-            mv nvim.appimage ./build/nvim
+            chmod +x ./.github/workflows/scripts/setup-appimage.sh
+            ./.github/workflows/scripts/setup-appimage.sh
```
Comment (Conni2461) [inline .github/workflows/ci.yml:54]: I think you misunderstood me. I meant making the script that lies in the repo executable, by running `chmod +x` for that script in your local repo and then pushing this change to make it permanent for this file.

File: .github/workflows/scripts/setup-appimage.bash (hunk 2 above)
Comment (Conni2461) [reply .github/workflows/scripts/setup-appimage.bash]: Just saying i never approved this PR, there is more missing than just linting that failed and thirdly. The logs where available but this job is now so old that github deleted the logs. I maintained lets say 95% of the PRs submitted to this repo since december and we not had once an issue with appimage not running. This also applies to other repositories we maintain like plenary.nvim or fzf-native, etc ... So my question again: It works perfectly fine why do we need a weird hacky workaround for an issue we dont have.

FYI this PR https://github.com/nvim-telescope/telescope.nvim/pull/980 was just submitted that tries to implement the same feature.

File: .github/workflows/ci.yml (hunk 4 above)
Comment (CyberGsus) [reply .github/workflows/ci.yml:54]: Oh, executable permissions persist when launching a container with a shared volume? Didn't know that, thanks!

**Comment by CyberGsus** [comment]
Done. I also rebased the branch to accept the changes from `master`

File: lua/telescope/make_entry.lua [hunk 5]
```diff
@@ -1111,52 +1111,5 @@ function make_entry.gen_from_git_status(opts)
   end
 end
 
-function make_entry.gen_from_jumplist(opts)
```
Comment (Conni2461) [inline lua/telescope/make_entry.lua:1114]: why is this here? Something went wrong with rebase

Comment (Conni2461) [reply lua/telescope/make_entry.lua:1114]: Also a bunch of other files

**Comment by Conni2461** [comment]
You still havent answered to this comment https://github.com/nvim-telescope/telescope.nvim/pull/970#discussion_r669016816

Also something went wrong with the rebase.

I am fine with merging your PR after the rebase issue are fixed and the appimage discussion is solved. Right now i still havent gotten a "good" justification. I am fine with either of those two. We remove appimage from ci (only run tests in x86 neovim and docgen now uses x86 neovim) or we remove that hacky workaround in this PR.

**Comment by CyberGsus** [comment]
Sorry for the inactivity. If it works fine currently, there's no point in making a workaround for it. I did it because I couldn't get logs from github for PRs that failed the lints, but that sounds more like a "me" problem. I'll close this PR, thanks for pointing it out.
